src/test/java/de/tum/in/www1/artemis/hestia/ProgrammingExerciseTaskServiceTest.java

```diff
@@ -0,0 +1,204 @@
+        });
+        assertThat(tasks).anyMatch(programmingExerciseTask -> {
+            var testCases = programmingExerciseTask.getTestCases();
+            return "Task 2".equals(programmingExerciseTask.getTaskName()) && !testCases.isEmpty()
+                    && "testMethods[Context]".equals(testCases.stream().findFirst().get().getTestName());
+        });
+        assertThat(tasks).anyMatch(programmingExerciseTask -> {
+            var testCases = programmingExerciseTask.getTestCases();
+            return "Task 3".equals(programmingExerciseTask.getTaskName()) && !testCases.isEmpty()
+                    && "testMethods[Policy]".equals(testCases.stream().findFirst().get().getTestName());
+        });
+
+        // Test that the other tasks were not removed and re-added.
+        var newTaskIds = tasks.stream().map(ProgrammingExerciseTask::getId).collect(Collectors.toSet());
+        assertThat(newTaskIds).containsAll(previousTaskIds);
+    }
+
+    @Test
+    public void testRemoveAllTasks() {
+        updateProblemStatement("Empty");
+        assertThat(programmingExerciseTaskRepository.findAll()).isEmpty();
+    }
+
+    @Test
+    public void testReduceToOneTask() {
+        updateProblemStatement("[task][Task 1](testClass[BubbleSort],testMethods[Context],testMethods[Policy])");
+        assertThat(programmingExerciseTaskRepository.findAll()).hasSize(1);
+        var tasks = programmingExerciseTaskRepository.findByExerciseIdWithTestCases(programmingExercise.getId());
+        assertThat(tasks).hasSize(1);
+        var task = tasks.stream().findFirst().get();
+        assertThat(task.getTaskName()).isEqualTo("Task 1");
+        assertThat(task.getTestCases()).hasSize(3);
+        assertThat(Set.of("testClass[BubbleSort]", "testMethods[Context]", "testMethods[Policy]"))
+                .isEqualTo(task.getTestCases().stream().map(ProgrammingExerciseTestCase::getTestName).collect(Collectors.toSet()));
+    }
+
+    /**
+     * Tests that renaming a task does not remove and read the task, but instead updates it
+     */
+    @Test
+    public void testRenameTask() {
+        var previousTaskIds = programmingExerciseTaskRepository.findByExerciseIdWithTestCases(programmingExercise.getId()).stream().map(ProgrammingExerciseTask::getId)
+                .collect(Collectors.toSet());
+
+        updateProblemStatement("""
+                [task][Task 1a](testClass[BubbleSort])
+                [task][Task 2](testMethods[Context])
+                """);
+
+        assertThat(programmingExerciseTaskRepository.findAll()).hasSize(2);
+        var tasks = programmingExerciseTaskRepository.findByExerciseIdWithTestCases(programmingExercise.getId());
+
+        var newTaskIds = tasks.stream().map(ProgrammingExerciseTask::getId).collect(Collectors.toSet());
+        assertThat(previousTaskIds).isEqualTo(newTaskIds);
+
+        assertThat(tasks).isEqualTo(programmingExerciseTaskRepository.findByExerciseIdWithTestCases(programmingExercise.getId()));
```

Comment:
```suggestion
        assertThat(programmingExerciseTaskRepository.findByExerciseIdWithTestCases(programmingExercise.getId())).isEqualTo(tasks);
```
Switched actual and expected.